chia/rpc/wallet_rpc_api.py

```diff
@@ -839,6 +840,15 @@ async def get_transaction(self, request: Dict) -> EndpointResult:
             "transaction_id": tr.name,
         }
 
+    async def get_transaction_memo(self, request: Dict) -> EndpointResult:
+        transaction_id: bytes32 = bytes32(hexstr_to_bytes(request["transaction_id"]))
+        tr: Optional[TransactionRecord] = await self.service.wallet_state_manager.get_transaction(transaction_id)
+        if tr is None:
+            raise ValueError(f"Transaction 0x{transaction_id.hex()} not found")
+        if tr.spend_bundle is None or len(tr.spend_bundle.coin_spends) == 0:
+            raise ValueError(f"Transaction 0x{transaction_id.hex()} doesn't have any coin spend.")
+        return {transaction_id.hex(): compute_memos(tr.spend_bundle)}
```

Comment:
the memo is the *list* of additional parameters, right? only the first item in that list is the *hint*. We have a database where the hint can be looked up (which probably would be more efficient than executing the CLVM code of the spend bundle). But in this case you really want the whole memo I take it.

Reply:
@arvidn  good point, thanks, @paninaro since memos are [hint, other memos ... ] you might want to think about how to display this in UX (if it matters)

Reply:
@ytx1991 afaik `get_transaction` only fetches transactions that user sent/created, but not for transactions received from other wallets?